sky/engine/core/dart/text.dart

```diff
@@ -514,23 +539,65 @@ class TextPosition {
   }
 }
 
+/// A paragraph of text.
+///
+/// A paragraph retains the size and position of each glyph in the text and can
+/// be efficiently resized and painted.
 abstract class Paragraph extends NativeFieldWrapperClass2 {
+  /// The minimum amount of horizontal space this paragraph of text is permitted to occupy.
   double get minWidth native "Paragraph_minWidth";
   void set minWidth(double value) native "Paragraph_setMinWidth";
+
+  /// The maximum amount of horizontal space this paragraph of text is permitted to occupy.
   double get maxWidth native "Paragraph_maxWidth";
   void set maxWidth(double value) native "Paragraph_setMaxWidth";
+
+  /// The minimum amount of vertical space this paragraph of text is permitted to occupy.
   double get minHeight native "Paragraph_minHeight";
   void set minHeight(double value) native "Paragraph_setMinHeight";
+
+  /// The maxium amount of vertical space this paragraph of text is permitted to occupy.
   double get maxHeight native "Paragraph_maxHeight";
   void set maxHeight(double value) native "Paragraph_setMaxHeight";
+
+  /// The amount of horizontal space this paragraph occupies.
+  ///
+  /// Valid only after [layout] has been called.
   double get width native "Paragraph_width";
+
+  /// The amount of vertical space this paragraph occupies.
+  ///
+  /// Valid only after [layout] has been called.
   double get height native "Paragraph_height";
+
+  /// The minimum width that this paragraph could be without failing to paint
+  /// its contents within itself.
+  ///
+  /// Valid only after [layout] has been called.
   double get minIntrinsicWidth native "Paragraph_minIntrinsicWidth";
+
+  /// Returns the smallest width beyond which increasing the width never
+  /// decreases the height.
+  ///
+  /// Valid only after [layout] has been called.
   double get maxIntrinsicWidth native "Paragraph_maxIntrinsicWidth";
+
+  /// The distance from the top of the paragraph to the first alphabetic baseline, in logical pixels.
```

Comment:
here and below, change "first foo baseline" to "foo baseline of the first line" or some such